keys-component.js

```diff
@@ -59,7 +66,11 @@ function factory($scope, $routeParams, brAlertService, brKeyService,
     self.modals.key = null;
   };
 
-  self.init = function(identity) {
+  self.onKeyGenerate = function(key) {
```

Comment:
Added some logging which definitely makes the occurrence of the issue much more rare.  I believe this definitely indicates a race condition.   With the extra FIXME logging, the problem may only occur one out of twenty attempts.  It's more like 1 out of 5 otherwise.